### ISSUE TITLE
Update UrlChecker to use proposed rst-parser event

### ISSUE THREAD

**Issue by weaverryan** [symfony-tools/docs-builder#16]
See: https://github.com/weaverryan/docs-builder/pull/2/files#r245185858

Basically, once/if/hopefully when doctrine/rst-parser#89 is merged, we can update our code to simplify things.

Also, to speed things up, we could:

A) Collect all of the links, but not actually test them
B) Once we have them all, test the links but in parallel (e.g. maybe 5 at a time). 

This would also allow us to have a reliable progress bar.

**Comment by javiereguiluz** [comment]
Similar to #8, let's close this one as "won't fix". We rarely link to sites different than php.net, github.com or wikipedia.org, so it's fine to not check all links. Readers will quickly tell us if some link is broken. Thanks!
